File: leo/G-9996.js
/**
 *  제목 : 한국 그리울 서버
 *  입력 : 100 * 100
 *  제한사항 : 1s
 *  시간계산 필요 : X
 */

// Run by Node.js
const readline = require('readline');

(async () => {
    const rl = readline.createInterface({ input: process.stdin });
    
    const input = [];
    for await (const line of rl) {
        input.push(line);
    }
    rl.close();

    const pattern = input.shift();
    const [prefix, suffix] = pattern.split('*');

    input.forEach(el => {
		if (el.length < prefix.length + suffix.length) {
            console.log('NE');
		} else if (el.startsWith(prefix) && el.endsWith(suffix)) {
            console.log('DA');
        } else {
            console.log('NE');
        }
    });

    process.exit();
})();

File: mindy/week1/G-9996.py
# ab*cde 인 경우도 고려하기
# a*a인 경우 a를 넣으면 틀림!

n = int(input())
pat = input().split("*")
length = len(pat[0])+len(pat[1])
for i in range(n):
    str = input()
    if length <= len(str):
        if str.startswith(pat[0]) and str.endswith(pat[1]):
            print('DA')
        else: print('NE')

    else: print('NE')
